### PR TITLE
Favourite tasks

### PR DESCRIPTION
**Pull request by wdoganowski**
Possibility to favourite the tasks

On the view with tasks, there are two additional lists and two icons: favourite and complete

**NOTE: It requires DB migration as new table is added: FavouriteTask. However the migration fails.** 

I have noticed that there is one DocumentedTask (id 15) without TaskApproval. But deleting it does not help. Problem is somewhere at the definition of TaskApproval as it has not default and does not allow Null, while probably there is some data like this (but I can't find it). Probably the TaskApproval for DocumentedTask 15 is hanging somewhere without the link (?)

I was not able to fix the migration.

### CODE REVIEW

**Comment by wdoganowski** [comment]
Ah, one more thing, when the task has short description, the task panel is not being displayed in full width. I don't know how to fix it.

image

File: harc_game_web/apps/tasks/views.py [hunk 1]
```diff
@@ -83,13 +98,36 @@ def complete_task(request):
...
     else:
         form = CompleteTaskForm(request)
     completed_tasks = DocumentedTask.objects.filter(user=request.user)
     return render(request, 'tasks/upload.html', {'form': form, 'completed_tasks': completed_tasks})
 
 
+@login_required
+def fav_task(request, id, tab=None):
+    """
+    Function to handle completing Task by Scout - render and process form
```
Comment (zkunik) [inline harc_game_web/apps/tasks/views.py]: opis się nie zgadza

File: harc_game_web/apps/tasks/views.py [hunk 2]
```diff
@@ -83,13 +98,36 @@ def complete_task(request):
...
+    if FavouriteTask.objects.filter(user=request.user, task=task).count()==0:
+        FavouriteTask.objects.create(user=request.user, task=task)
+
+    return redirect(reverse('tasks', args={tab}))
+
+
+@login_required
+def unfav_task(request, id, tab=None):
+    """
+    Function to handle completing Task by Scout - render and process form
```
Comment (zkunik) [inline harc_game_web/apps/tasks/views.py]: opis się nie zgadza

File: harc_game_web/apps/tasks/templates/tasks/view.html [hunk 3]
```diff
@@ -19,23 +19,47 @@
...
 {% block content %}
 
+    <table>
+        <tr>
+    <div class="rpgui-icon potion-green" style="float:right; margin-top:5px; margin-right:35px;"></div>
+    <div style="float:right; margin-top:5px; margin-right:5px;"><p>Ulubione</p></div>
+        </tr>
+        <tr>
+    <div class="rpgui-icon potion-red" style="float:right; margin-top:5px; margin-right:15px;"></div>
+    <div style="float:right; margin-top:5px; margin-right:5px;"><p>Wykonane</p></div>
```
Comment (zkunik) [inline harc_game_web/apps/tasks/templates/tasks/view.html]: pozbyłabym się tego "wykonane" i oznaczania zadań w taki sposób chyba

**Review by zkunik — APPROVED** [review]

**Pull request merged**